pypeit/core/coadd.py

```diff
@@ -883,103 +924,106 @@ def sn_weights(waves, fluxes, ivars, masks, sn_smooth_npix, const_weights=False,
-        for iexp in range(nexp):
-            sensfunc_iord = flux_calib.get_sensfunc_factor(waves_stack[:, iord, iexp], wave_zp[:, iord], zeropoint[:, iord], 1.0, extrap_sens=extrap_sens)
-            weights_stack[:, iord, iexp] = utils.inverse(sensfunc_iord)
-
-    if debug:
-        weights_qa(waves_stack, weights_stack, (waves_stack > 1.0), title='sensfunc_weights')
-
-    if waves.ndim == 2:
-        weights_stack = np.reshape(weights_stack, (nspec, norder))
-
-    return weights_stack
-
-# TODO Rename this function to something sensfunc related
-def get_tell_from_file(sensfile, waves, masks, iord=None):
-    '''
-    Get the telluric model from the sensfile.
-
-    Args:
-        sensfile (str): the name of your fits format sensfile
-        waves (ndarray): wavelength grid for your output telluric model
-        masks (ndarray, bool): mask for the wave
-        iord (int or None): if None returns telluric model for all orders, otherwise return the order you want
-
-    Returns:
-         ndarray: telluric model on your wavelength grid
-    '''
-
-
-    sens_param = Table.read(sensfile, 1)
-    sens_table = Table.read(sensfile, 2)
-    telluric = np.zeros_like(waves)
-
-    if (waves.ndim == 1) and (iord is None):
-        msgs.info('Loading Telluric from Longslit sensfiles.')
-        tell_interp = scipy.interpolate.interp1d(sens_table[0]['WAVE'], sens_table[0]['TELLURIC'], kind='cubic',
-                                        bounds_error=False, fill_value=np.nan)(waves[masks])
-        telluric[masks] = tell_interp
-    elif (waves.ndim == 1) and (iord is not None):
-        msgs.info('Loading order {:} Telluric from Echelle sensfiles.'.format(iord))
-        wave_tell_iord = sens_table[iord]['WAVE']
-        tell_mask = (wave_tell_iord > 1.0)
-        tell_iord = sens_table[iord]['TELLURIC']
-        tell_iord_interp = scipy.interpolate.interp1d(wave_tell_iord[tell_mask], tell_iord[tell_mask], kind='cubic',
-                                        bounds_error=False, fill_value=np.nan)(waves[masks])
-        telluric[masks] = tell_iord_interp
-    else:
-        norder = np.shape(waves)[1]
-        for iord in range(norder):
-            wave_iord = waves[:, iord]
-            mask_iord = masks[:, iord]
-
-            # Interpolate telluric to the same grid with waves
-            # Since it will be only used for plotting, I just simply interpolate it rather than evaluate it based on the model
-            wave_tell_iord = sens_table[iord]['WAVE']
-            tell_mask = (wave_tell_iord > 1.0)
-            tell_iord = sens_table[iord]['TELLURIC']
-            tell_iord_interp = scipy.interpolate.interp1d(wave_tell_iord[tell_mask], tell_iord[tell_mask], kind='cubic',
-                                                    bounds_error=False, fill_value=np.nan)(wave_iord[mask_iord])
-            telluric[mask_iord, iord] = tell_iord_interp
-
-    return telluric
+# NOTE: This was moved to pypeit/sensfunc.py to avoid a circular import
```

Comment:
Is there another way to avoid such a circular import (which was not occurring before). This is a core routine, which is why I ask. Other modules use sensfunc_weights.

Reply:
Not that I could find.  The main issue is the `pypeit/sensfunc.py` now imports `pypeit.core.telluric`, which wreaked havoc.  Moving this (and a few other) functions was the easiest solution.